overlord/snapstate/catalogrefresh.go

```diff
@@ -57,10 +61,14 @@ func (r *catalogRefresh) Ensure() error {
 	}
 
 	now := time.Now()
+	catalogExists := true
```

Comment:
I find this flag name and the default a bit confusing, we actually check  for existence only after restart, but if we didn't manage to create it will not magically exist the next time we are here. I don't think this provokes serious bugs but is confusing to follow.   I think we should actually pick the delay earlier because really the delay is not about whether the catalog exists but whether we are the fist time here after a restart or not.

Reply:
Sorry for applying a change related to this on my own.